src/beast/beast/nudb/detail/format.h

```diff
@@ -91,23 +97,65 @@ struct log_file_header
     static std::size_t BEAST_CONSTEXPR size =
         8 +     // Type
         2 +     // Version
+        8 +     // UID
         8 +     // Appnum
+        2 +     // KeySize
+
         8 +     // Salt
         8 +     // Pepper
-        2 +     // KeySize
+        2 +     // BlockSize
+
         8 +     // KeyFileSize
         8;      // DataFileSize
 
     char type[8];
     std::size_t version;
+    std::uint64_t uid;
     std::uint64_t appnum;
+    std::size_t key_size;
     std::uint64_t salt;
     std::uint64_t pepper;
-    std::size_t key_size;
+    std::size_t block_size;
     std::size_t key_file_size;
     std::size_t dat_file_size;
 };
 
+// Type used to store hashes in buckets.
+// This can be smaller than the output
+// of the hash function.
+//
+using hash_t = uint48_t;
+
+static_assert(field<hash_t>::size <=
+    sizeof(std::size_t), "");
+
+template <class T>
+std::size_t
+make_hash (std::size_t h);
+
+template<>
+inline
+std::size_t
+make_hash<uint48_t>(std::size_t h)
```

Comment:
@seelabs That's not right, the intent is for a family of `make_hash` functions so that `hash_type` can become a template argument in the `store` class.